src/com/palmergames/util/FileMgmt.java

```diff
@@ -578,6 +587,74 @@ public static void deleteUnusedFiles(File residentDir, Set<String> fileNames) {
 		}
 	}
 	
+	public static void writeRegenData(Collection<PlotBlockData> data, File file) {
```

Comment:
Since both the regen data and snapshot data utilize collections of lines couldn't you use the `listToFile` method?
E.g. in the datasource class to save regen something like.
```java
Collection<String> lines = data.stream()
.map(plot-> plot.getWorldName() + "," + plot.getX() + "," + plot.getZ())
.collect(Collectors.toList());
FileMgmt.listToFile(lines, sourcePath);
```
I'm just trying to come up with a way to abstract FileMgmt so that it can be used as general as possible and so that other classes don't have touch the file streams directly if not neccesary.